client/client_inventory.go

```diff
@@ -35,29 +35,43 @@ func NewInventory() InventorySubmitter {
 
 // Submit reports status information to the backend
 func (i *InventoryClient) Submit(api ApiRequester, url string, data interface{}) error {
-	req, err := makeInventorySubmitRequest(url, data)
+	// PATCH used to be the only method available in Mender Product 2.5, so
+	// fall back to that if PUT fails.
+	r, err := doSubmitInventory(api, http.MethodPut, url, data)
+	if r != nil && r.StatusCode == http.StatusMethodNotAllowed {
+		r, err = doSubmitInventory(api, http.MethodPatch, url, data)
```

Comment:
Nit:
```suggestion
		_, err = doSubmitInventory(api, http.MethodPatch, url, data)
```
I don't see this used anywhere :eyes:

Reply:
I think this should stay. If more code is added after this, `r` might be assigned the wrong result if `_` is used in the second call.